src/scan.c

```diff
@@ -179,8 +181,11 @@ op_t parseConstant(const char **iter) {
 
 static void scanChar(const char **iter, char expected) {
     for (char ch; (ch = **iter) != expected; (*iter)++) {
+        if (ch == '\n') {
+            lineNumber++;
+        }
         if (!shouldIgnore(ch)) {
-            fprintf(stderr, "When seeking %c found unexpected character %c, before: %s", expected, ch, *iter);
```

Comment:
huge improvement for developer experience since this would previously dump the whole remaining file